compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1343,6 +1350,8 @@ protected ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLo
             case KEY_SPECIFIER:
             case ERROR_TYPE_DESCRIPTOR:
             case LET_VAR_DECL:
+            case NEW_RHS:
+            case EXPLICIT_NEW_RHS:
```

Comment:
You are starting a context for `NEW_RHS` and `EXPLICIT_NEW_RHS` here, but its never ended.

Reply:
fixed with 07c06fd